src/db/index.js

```diff
@@ -41,13 +44,41 @@ const dbFactory = opts => {
   }
 
   // queryS :: (String, [a]) -> Stream b
-  const queryS = (sql, vals=[]) =>
-    _((async () => {
-      const client = await pool.connect()
-      const stream = _(client.query(new QueryStream(sql, vals)))
-      stream.observe().done(() => client.release())
-      return stream
-    })()).flatten()
+  const queryS = (sql, vals=[]) => {
+    let client, cursor
+
+    const close = once(err1 => {
+      debug(`closing cursor${err1 ? ' on error' : ''}`)
+      cursor.close(err2 =>
+        client.release(err1 || err2)
+      )
+    })
+
+    const openCursor = klient => {
+      debug('opening cursor')
+      client = klient
+      cursor = client.query(new Cursor(sql, vals))
+
+      return _((push, next) => {
+        cursor.read(100, (err, rows) => {
+          if (err) {
+            push(err)
+            next()
+          } else if (rows.length) {
+            push(null, rows)
+            next()
+          } else {
+            push(null, _.nil)
+          }
+        })
```

Comment:
:thinking: 

```js
cursor.read(100, (err, rows) => {
  push(err, rows)
  if (rows.length > 0) next()
})
```

Reply:
The `rows` is empty when the cursor has reached the end, and I didn't know how appropriate it was to push an empty array.  I know it's `flatten`ed, but... still.

Also, I still need to push a `_.nil` when the cursor is ended, right?  And is it appropriate to push a `_.nil` after I've potentially pushed an `err`?

Maybe more like this?
```js
cursor.read(100, (err, rows) => {
  push(err, rows)
  if (rows.length) next()
  else push(null, _.nil)
})
```

Reply:
Yup, you're right. That's better.